server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java

```diff
@@ -245,23 +247,16 @@ private int getRandomPriority() {
 
     private void assertNoRunningTasks(PrioritizedThrottledTaskRunner<TestTask> taskRunner) {
         logger.info("--> ensure that there are no running tasks in the executor. Max number of threads [{}]", maxThreads);
-        final var barrier = new CyclicBarrier(maxThreads + 1);
+        final var barrier = new TestBarrier(maxThreads + 1);
         for (int i = 0; i < maxThreads; i++) {
             executor.execute(() -> {
                 logger.info("--> await until barrier is released");
-                awaitBarrier(barrier);
+                barrier.await(10, TimeUnit.SECONDS);
                 logger.info("--> the barrier is released");
             });
         }
-        awaitBarrier(barrier);
+        barrier.await(10, TimeUnit.SECONDS);
         assertThat(taskRunner.runningTasks(), equalTo(0));
     }
 
-    private static void awaitBarrier(CyclicBarrier barrier) {
-        try {
-            barrier.await(10, TimeUnit.SECONDS);
-        } catch (Exception e) {
-            throw new AssertionError("unexpected", e);
-        }
-    }
```

Comment:
Aside from getting rid of this, is there any other advantage in introducing `TestBarrier`?

Reply:
No, it's just about those pointless checked exceptions.

Reply:
(I can do this separately too, there's lots of other spots where this is useful)

Reply:
TBH, I find the class unnecessary. A utility function which takes a timeout parameter will do as well. But I'm fine with leaving it, if you'd like.

Reply:
Ok, I'll do this in a separate PR

Reply:
If you find it useful, it's perfectly fine here. I was just trying to understand why we need it at all. 😄